### PR TITLE
Reintroduce tenantcluster 

### PR DESCRIPTION
**Pull request by tomahawk28**
Following https://github.com/giantswarm/statusresource/pull/40#issuecomment-563513451

> The way it is after this PR, there's no way to construct K8sClient and configure it to StatusResource

Thanks @sslavic

### CODE REVIEW

File: create.go [hunk 1]
```diff
@@ -217,45 +220,79 @@ func (r *Resource) computeCreateEventPatches(ctx context.Context, obj interface{
...
+			i, err := r.clusterIDFunc(obj)
+			if err != nil {
+				return nil, microerror.Mask(err)
+			}
+			e, err := r.clusterEndpointFunc(obj)
+			if err != nil {
+				return nil, microerror.Mask(err)
+			}
+
+			restConfig, err := r.tenantCluster.NewRestConfig(ctx, i, e)
```
Comment (tomahawk28) [inline create.go]: `create.go` is the same as two commits before the current master. Only `tenantCluster` here use a different function.

**Review by sslavic — APPROVED** [review]
LGTM

File: Gopkg.lock [hunk 2]
```diff
@@ -138,6 +147,14 @@
   pruneopts = "NUT"
   revision = "5d098618662e74384f52bf064fe3588436873eb2"
 
+[[projects]]
+  branch = "master"
+  digest = "1:07185f49b13d696336e00ccb9e4e009137eb5fb42af2c406a827b547a527e4f5"
+  name = "github.com/giantswarm/tenantcluster"
```
Comment (sslavic) [inline Gopkg.lock]: Please add it to Gopkg.toml as explicit dependency locked ot master branch

**Review by rossf7 — APPROVED** [review]
LGTM

**Pull request merged**